sshauthentication-api/src/main/java/org/openintents/ssh/authentication/SshAuthenticationApi.java

```diff
@@ -0,0 +1,174 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.openintents.ssh.authentication;
+
+import android.annotation.TargetApi;
+import android.content.Context;
+import android.content.Intent;
+import android.os.AsyncTask;
+import android.os.Build;
+import android.util.Log;
+import org.openintents.ssh.authentication.ISshAuthenticationService;
+
+public class SshAuthenticationApi {
+    private static final String TAG = "SshAuthenticationApi";
+
+    public static final String SERVICE_INTENT = "org.openintents.ssh.ISshAuthenticationService";
+
+    public static final int API_VERSION = 1;
+    public static final String EXTRA_API_VERSION = "api_version";
+
+    public static final int RESULT_CODE_ERROR = 0;
+    public static final int RESULT_CODE_SUCCESS = 1;
+    public static final int RESULT_CODE_USER_INTERACTION_REQUIRED = 2;
+
+    public static final String EXTRA_ERROR = "error";
+
+    /**
+     * Global
+     */
+    public static final String ACTION_SIGN = "org.openintents.ssh.action.SIGN";
+    public static final String ACTION_SELECT_KEY = "org.openintents.ssh.action.SELECT_KEY";
+    public static final String ACTION_GET_PUBLIC_KEY = "org.openintents.ssh.action.GET_PUBLIC_KEY";
+
+    public static final String EXTRA_RESULT_CODE = "result_code";
+    public static final String EXTRA_PENDING_INTENT = "intent";
+
+    public static final String EXTRA_KEY_ID = "key_id";
+
+    /**
+     * ACTION_SIGN
+     */
+    public static final String EXTRA_CHALLENGE = "challenge";
+    public static final String EXTRA_HASH_ALGORITHM = "hash_algorithm";
+    public static final String EXTRA_SIGNATURE = "signature";
+
+    /**
+     * ACTION_SELECT_KEY
+     */
+    public static final String EXTRA_KEY_DESCRIPTION = "key_description";
+
+    /**
+     * ACTION_GET_PUBLIC_KEY
+     */
+    public static final String EXTRA_PUBLIC_KEY = "public_key";
+    public static final String EXTRA_PUBLIC_KEY_ALGORITHM = "public_key_algorithm";
+    public static final String EXTRA_PUBLIC_KEY_FORMAT = "public_key_format";
+
+    /* hash algorithms used for signature generation */
+    public static final int SHA1 = 0;
+    public static final int SHA224 = 1;
+    public static final int SHA256 = 2;
+    public static final int SHA384 = 3;
+    public static final int SHA512 = 4;
+    public static final int RIPEMD160 = 5;
+
+    /* signature algorithms */
+    public static final int RSA = 0;
+    public static final int ECDSA = 1;
+    public static final int EDDSA = 2;
+    // TODO: remove?
+    public static final int DSA = 3;
```

Comment:
I can't decide whether to keep DSA support in the library. Besides the fact that OpenKeychain doesn't support DSA and OpenSSH has it disabled in default config, I see no reasons to not support it.

Reply:
I think we still support DSA keys if you import them, but we don't support generation of DSA keys/subkeys. IMHO You can include it.